### PR TITLE
Sync subliminal lib with upstream 2.1.0-dev

### PR DESCRIPTION
**Pull request by fernandog**
TODO:
- [x] podnapisi as https. no idea how

UPDATE:
Removed that commit about ltv archived cache removal

### CODE REVIEW

File: lib/subliminal/providers/legendastv.py [hunk 1]
```diff
@@ -143,6 +143,9 @@ def get_matches(self, video, hearing_impaired=False):
             if video.imdb_id and self.imdb_id == video.imdb_id:
                 matches.add('imdb_id')
 
+        # archive name
```
Comment (ratoaq2) [inline lib/subliminal/providers/legendastv.py]: This was already removed upstream
https://github.com/Diaoul/subliminal/blob/develop/subliminal/providers/legendastv.py

File: lib/subliminal/providers/itasa.py [hunk 2]
```diff
@@ -1,482 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (ratoaq2) [inline lib/subliminal/providers/itasa.py]: You should add itasa in our code

File: tests/test_subtitles.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 def test_sorted_service_list(monkeypatch):
     # Given
-    monkeypatch.setattr(app, 'SUBTITLES_SERVICES_LIST', ['legendastv', 'trash', 'itasa', 'thesubdb', 'shooter'])
+    monkeypatch.setattr(app, 'SUBTITLES_SERVICES_LIST', ['legendastv', 'trash', 'thesubdb', 'shooter'])
```
Comment (ratoaq2) [inline tests/test_subtitles.py]: You shouldn't change the test. 
The same functionality should continue to work

File: lib/subliminal/providers/itasa.py (hunk 2 above)
Comment (fernandog) [inline lib/subliminal/providers/itasa.py]: @ratoaq2 where exactly to add the file?

File: lib/subliminal/utils.py [hunk 4]
```diff
@@ -6,20 +6,6 @@
 import struct
 
 
-def hash_itasa(video_path):
```
Comment (ratoaq2) [inline lib/subliminal/utils.py]: This also need to be added somewhere in medusa to allow itasa to work properly

File: lib/subliminal/providers/podnapisi.py [hunk 5]
```diff
@@ -82,7 +82,7 @@ class PodnapisiProvider(Provider):
     """Podnapisi Provider."""
     languages = ({Language('por', 'BR'), Language('srp', script='Latn')} |
                  {Language.fromalpha2(l) for l in language_converters['alpha2'].codes})
-    server_url = 'http://podnapisi.eu/subtitles/'
```
Comment (ratoaq2) [inline lib/subliminal/providers/podnapisi.py]: This can be changed in our subtitles.py

File: lib/subliminal/video.py [hunk 6]
```diff
@@ -13,7 +13,7 @@
                     '.bix', '.box', '.cam', '.dat', '.divx', '.dmf', '.dv', '.dvr-ms', '.evo', '.flc', '.fli',
                     '.flic', '.flv', '.flx', '.gvi', '.gvp', '.h264', '.m1v', '.m2p', '.m2ts', '.m2v', '.m4e',
                     '.m4v', '.mjp', '.mjpeg', '.mjpg', '.mkv', '.moov', '.mov', '.movhd', '.movie', '.movx', '.mp4',
-                    '.mpe', '.mpeg', '.mpg', '.mpv', '.mpv2', '.mxf', '.nsv', '.nut', '.ogg', '.ogm' '.ogv', '.omf',
+                    '.mpe', '.mpeg', '.mpg', '.mpv', '.mpv2', '.mxf', '.nsv', '.nut', '.ogg', '.ogm', '.ogv', '.omf',
```
Comment (ratoaq2) [inline lib/subliminal/video.py]: This can also be patched from our subtitles.py

Comment (fernandog) [inline lib/subliminal/video.py]: @ratoaq2 why this need to be patched? The comma was missing and it was added

File: medusa/subtitle_providers/legendastv.py [hunk 7]
```diff
@@ -200,7 +200,8 @@ def terminate(self):
 
         self.session.close()
 
-    @region.cache_on_arguments(expiration_time=SHOW_EXPIRATION_TIME)
```
Comment (fernandog) [inline medusa/subtitle_providers/legendastv.py]: @ratoaq2 or do we remove of we decrease `SHOW_EXPIRATION_TIME`

File: medusa/subtitle_providers/legendastv.py [hunk 8]
```diff
@@ -457,7 +457,14 @@ def list_subtitles(self, video, languages):
...
 
-        return [s for l in languages for s in self.query(l, title, season=season, episode=episode, year=video.year)]
+        found_subtitles = [s for l in languages for s in self.query(l, title, season=season, episode=episode, year=video.year)]
+        if not found_subtitles and 'marve' in title.lower():
+            title = title.lower().replace("marvel's ", '').replace('marvels ', '')
+            found_subtitles = [s for l in languages for s in self.query(l, title, season=season, episode=episode, year=video.year)]
+        if not found_subtitles and 'legends of tomorrow' in title.lower():
+            title = 'dc’s legends of tomorrow'
+            found_subtitles = [s for l in languages for s in self.query(l, title, season=season, episode=episode, year=video.year)]
+        return found_subtitles
```
Comment (fernandog) [inline medusa/subtitle_providers/legendastv.py]: @ratoaq2 the problem here is that show was inserted in ltv database with this `’`  and not `'`

maybe to fix for other shows we could replace `’`  with `'`

File: medusa/subtitle_providers/legendastv.py [hunk 9]
```diff
@@ -0,0 +1,490 @@
...
+        :param str title: the title to search for.
+        :return: found titles.
+        :rtype: dict
+
+        """
+        # make the query
+        logger.info('Searching title %r', title)
+        r = self.session.get(self.server_url + 'legenda/sugestao/{}'.format(title), timeout=10)
+        r.raise_for_status()
+        if self.is_under_maintaince(r):
```
Comment (fernandog) [inline medusa/subtitle_providers/legendastv.py]: Yes. In all raise for status. 
The maintaince can help in any page request

**Comment by fernandog** [comment]
@ratoaq2 btw still to make podnapisi as https. no idea how and where. help?

**Comment by fernandog** [comment]
@ratoaq2 can we make this to this release?

**Comment by fernandog** [comment]
@ratoaq2 will merge this now and then you add the https to podnapisi ok ?

File: medusa/init/__init__.py [hunk 10]
```diff
@@ -109,7 +110,28 @@ def _strptime_workaround():
...
 def _configure_guessit():
-    """Replace guessit function with a pre-configured one, so guessit.guessit() could be called directly in any place."""
+    """Replace guessit with a pre-configured one, so guessit.guessit() could be called directly in any place."""
     import guessit
     from ..name_parser.guessit_parser import guessit as pre_configured_guessit
     guessit.guessit = pre_configured_guessit
+
+
+def _configure_subliminal():
+    """Load subliminal with our custom configuration."""
```
Comment (fernandog) [inline medusa/init/__init__.py]: @ratoaq2 can we override `SHOW_EXPIRATION_TIME` here from 3 weeks to 3 days or else?

**Pull request merged**